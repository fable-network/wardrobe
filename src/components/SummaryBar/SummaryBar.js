import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Stack from '../../layout/Stack';

const StackHorizontal = props => <Stack direction="horizontal" size="large" {...props} />;

const Wrapper = styled(StackHorizontal)`
  padding: 1rem 0;
`;

const Buttons = styled(StackHorizontal)`
  margin-left: auto;
`;

const SummaryBar = ({ buttons, children }) => (
  <Wrapper>
    {children}
    {Boolean(buttons) && <Buttons>{buttons}</Buttons>}
  </Wrapper>
);

SummaryBar.defaultProps = {};

SummaryBar.propTypes = {
  children: PropTypes.node,
  buttons: PropTypes.node,
};

export default SummaryBar;
