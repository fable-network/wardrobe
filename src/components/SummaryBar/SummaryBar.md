SummaryBar component:

```jsx
const DisplayValue = require('../DisplayValue').default;
const Button = require('../Button').default;
<SummaryBar
  buttons={
    <React.Fragment>
      <Button>Clear cart</Button>
      <Button appearance="primary">Checkout</Button>
    </React.Fragment>
  }
>
  <DisplayValue label="Styles">3</DisplayValue>
  <DisplayValue label="Pre-packs">1</DisplayValue>
  <DisplayValue label="Pieces">10</DisplayValue>
</SummaryBar>
```
