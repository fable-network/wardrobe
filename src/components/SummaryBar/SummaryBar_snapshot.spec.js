/* eslint-disable */
// This file was automatically generated
import React from 'react';
import { shallow } from 'enzyme';
import SummaryBar from './SummaryBar';

describe('SummaryBar Component', () => {
  describe('snapshots', () => {
    it('should match snapshot', () => {
    
      const wrapper1 = shallow(
        <SummaryBar />
      );
      expect(wrapper1).toMatchSnapshot();

    });
  });
});
