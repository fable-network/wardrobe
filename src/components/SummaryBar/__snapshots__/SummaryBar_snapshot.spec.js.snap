// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SummaryBar Component snapshots should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SummaryBar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        undefined,
        false,
      ],
    },
    "ref": null,
    "rendered": Array [
      undefined,
      false,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          undefined,
          false,
        ],
      },
      "ref": null,
      "rendered": Array [
        undefined,
        false,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
